paddle/fluid/operators/conv_cudnn_op.cu.cc

```diff
@@ -361,263 +205,83 @@ class CUDNNConvGradOpKernel : public framework::OpKernel<T> {
-      VLOG(2) << "Cudnn workspace size bwd: "
-              << static_cast<double>(workspace_size_in_bytes) / (1 << 20)
-              << " MB";
+      // ------------------- cudnn descriptors ---------------------
+      filter_grad_data = filter_grad->mutable_data<T>(ctx.GetPlace());
+      args2.handle = handle;
+      args2.idesc.set(*input, iwo_groups);
+      args2.wdesc.set(*filter_grad, layout_tensor, iwo_groups);
+      args2.odesc.set(*output_grad, iwo_groups);
+      args2.cdesc.set(dtype, paddings, strides, dilations, c_groups);
+
+      using search2 = SearchAlgorithm<cudnnConvolutionBwdFilterAlgoPerf_t>;
+      filter_algo =
+          search2::Find<T>(args2, exhaustive_search, deterministic, 1, ctx);
+      workspace_size = std::max(workspace_size,
+                                search2::GetWorkspaceSize(args2, filter_algo));
     }
 
     // ------------------- cudnn conv backward data ---------------------
     ScalingParamType<T> alpha = 1.0f, beta = 0.0f;
     if (input_grad) {
-      T* input_grad_data = input_grad->mutable_data<T>(ctx.GetPlace());
       // Because beta is zero, it is unnecessary to reset input_grad.
-
       for (int i = 0; i < groups; i++) {
-        CUDNN_ENFORCE(platform::dynload::cudnnConvolutionBackwardData(
-            handle, &alpha, cudnn_filter_desc,
-            filter_data + i * group_offset_filter, cudnn_output_grad_desc,
-            output_grad_data + i * group_offset_out, cudnn_conv_desc, data_algo,
-            cudnn_workspace_ptr, workspace_size_in_bytes, &beta,
-            cudnn_input_desc, input_grad_data + i * group_offset_in));
+        workspace_handle.RunFunc(
+            [&](void* cudnn_workspace_ptr) {
+              CUDNN_ENFORCE(platform::dynload::cudnnConvolutionBackwardData(
+                  handle, &alpha, args1.wdesc.desc(),
+                  filter_data + i * group_offset_filter, args1.odesc.desc(),
+                  output_grad_data + i * group_offset_out, args1.cdesc.desc(),
+                  data_algo, cudnn_workspace_ptr, workspace_size, &beta,
+                  args1.idesc.desc(), input_grad_data + i * group_offset_in));
+            },
+            workspace_size);
       }
     }
     // ------------------- cudnn conv backward filter ---------------------
     if (filter_grad) {
-      T* filter_grad_data = filter_grad->mutable_data<T>(ctx.GetPlace());
       // Because beta is zero, it is unnecessary to reset filter_grad.
       for (int i = 0; i < groups; i++) {
-        CUDNN_ENFORCE(platform::dynload::cudnnConvolutionBackwardFilter(
-            handle, &alpha, cudnn_input_desc, input_data + i * group_offset_in,
-            cudnn_output_grad_desc, output_grad_data + i * group_offset_out,
-            cudnn_conv_desc, filter_algo, cudnn_workspace_ptr,
-            workspace_size_in_bytes, &beta, cudnn_filter_desc,
-            filter_grad_data + i * group_offset_filter));
+        workspace_handle.RunFunc(
```

Comment:
Here use `workspace_handle.RunFunc` not  ctx.AllocateTmpTensor to get cuDNN worksapce,  please ask @chengduoZH @sneaxiy to review.